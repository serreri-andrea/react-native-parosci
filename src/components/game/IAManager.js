/**
 * Created by Andrea on 18/03/2018.
 */

import React, { Component } from 'react';
import {
    StyleSheet,
    Text,
    View,
    FlatList,
    Image,
    TouchableHighlight
} from 'react-native';

import Answers              from "../../lib/Answers";
import Sizes                from "../../theme/sizes";

export default class IAManager extends Component{

    constructor(props) {
        super(props);
        this.state = {};
    }

    componentDidMount() {
        this.timeout = setInterval(() => {
            let answer = Answers.answersIAEasy(this.props.mode);
            this.setState({card: answer});
            this.handleChoosedCard(answer);
        }, 2000)
    }

    componentWillReceiveProps(props){
    }


    handleChoosedCard(card){
        if (this.props.score && (this.props.score.p1Score >= 3 || this.props.score.p2Score>= 3)){
            clearInterval(this.timeout);
            clearTimeout(this.timeout);
        }else {
            if (this.props && this.props.callback && typeof this.props.callback === "function")
             this.props.callback(card)
        }
    }

    renderCards(item){
        if (item && item.item) {
            return (
                <View>
                    <Image source={this.state.card === item.item.reference ? item.item.images.selected : item.item.images.default}
                           style={{height:Sizes.screen.height / 8, width:Sizes.screen.width / 5, margin:10}} />
                </View>
            )
        }else{return(<View/>)}
    }

    render(){
        return (
            <View style={styles.container}>
                <FlatList
                    data={this.props.cards}
                    extraData={this.state}
                    numColumns={3}
                    columnWrapperStyle={styles.container}
                    ItemSeparatorComponent={() => <View style={{margin:0}}/>}
                    renderItem={this.renderCards.bind(this)}
                    keyExtractor={(item, index) => index.toString()}/>
            </View>
        )
    }
}

const styles = StyleSheet.create({
    container:{
        marginTop:10,
        marginBottom:10,
        justifyContent:'center',
        alignItems:"center",
    },
});
