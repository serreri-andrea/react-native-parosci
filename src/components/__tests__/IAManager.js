import 'react-native';
import React            from 'react';
import IAManager        from '../game/IAManager';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly: without props', () => {
    const tree = renderer.create(
        <IAManager/>
    );
    expect(tree).toMatchSnapshot();
});

it('renders correctly: without props', () => {
    const tree = renderer.create(
        <IAManager cards={"frghjkl"} mode={"fghjklm"} callback={"dfghjukilm"} score={"fghjklm"}/>
    );
    expect(tree).toMatchSnapshot();
});


