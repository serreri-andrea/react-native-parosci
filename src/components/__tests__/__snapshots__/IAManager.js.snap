// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly: with props 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "center",
      "marginBottom": 10,
      "marginTop": 10,
    }
  }
>
  <RCTScrollView
    ItemSeparatorComponent={[Function]}
    data="frghjkl"
    disableVirtualization={false}
    extraData={Object {}}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={true}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View>
          <Image
            source={undefined}
            style={
              Object {
                "backgroundColor": "pink",
                "height": 50,
                "width": 50,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "margin": 20,
            }
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View>
          <Image
            source={undefined}
            style={
              Object {
                "backgroundColor": "pink",
                "height": 50,
                "width": 50,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "margin": 20,
            }
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View>
          <Image
            source={undefined}
            style={
              Object {
                "backgroundColor": "pink",
                "height": 50,
                "width": 50,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "margin": 20,
            }
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View>
          <Image
            source={undefined}
            style={
              Object {
                "backgroundColor": "pink",
                "height": 50,
                "width": 50,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "margin": 20,
            }
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View>
          <Image
            source={undefined}
            style={
              Object {
                "backgroundColor": "pink",
                "height": 50,
                "width": 50,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "margin": 20,
            }
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View>
          <Image
            source={undefined}
            style={
              Object {
                "backgroundColor": "pink",
                "height": 50,
                "width": 50,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "margin": 20,
            }
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View>
          <Image
            source={undefined}
            style={
              Object {
                "backgroundColor": "pink",
                "height": 50,
                "width": 50,
              }
            }
          />
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`renders correctly: without props 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "center",
      "marginBottom": 10,
      "marginTop": 10,
    }
  }
>
  <RCTScrollView
    ItemSeparatorComponent={[Function]}
    data={undefined}
    disableVirtualization={false}
    extraData={Object {}}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={true}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;
